godot-core/src/init/mod.rs

```diff
@@ -119,6 +120,11 @@ unsafe extern "C" fn ffi_deinitialize_layer<E: ExtensionLibrary>(
     });
 }
 
+#[doc(hidden)]
+// FIXME: this is terribly terrible.
+// this is appended to by the proc macro.
+pub static DOCS: Mutex<Vec<String>> = Mutex::new(Vec::new());
+
```

Comment:
we already have a way to do things like this, using [the plugin system](https://github.com/godot-rust/gdext/blob/master/godot-core/src/registry/plugin.rs).

Reply:
im afraid i dont know how to use it. could you explain?

Reply:
Here's an example of how a class plugin is registered by the `GodotClass` derive macro:
```rs
::godot::sys::plugin_add!(__GODOT_PLUGIN_REGISTRY in::godot::private;
::godot::private::ClassPlugin {
    class_name: ...,
    item: ::godot::private::PluginItem::Struct { ... });
```
This `plugin_add!` macro will ensure that the `ClassPlugin` here is put into a global array `__GODOT_PLUGIN_REGISTRY`. Thus we can later access it during init, which happens [here](https://github.com/godot-rust/gdext/blob/529654cf2dec8f7d6670155b058aa36927bbf4d6/godot-core/src/registry/class.rs#L157).

Usually you can just change the pre-existing plugins a bit, and do the stuff you need with them during library initialization like above.

Reply:
alright that worked well, sorry i missed this comment.